platform/darwin/src/MGLStyle_Private.hpp

```diff
@@ -0,0 +1,11 @@
+#import "MGLStyle.h"
+
+#import "MGLStyleLayer.h"
+#import "MGLFillStyleLayer.h"
+#import <mbgl/util/default_styles.hpp>
+#include <mbgl/mbgl.hpp>
+
+@interface MGLStyle (Private)
+@property (nonatomic, weak) MGLMapView *mapView;
+@property (nonatomic) mbgl::Map *mbglMap;
```

Comment:
Let’s avoid keeping pointer references to `mbgl::Map` outside of MGLMapView. MGLMapView should have a private method that returns the `mbgl::Map`, or perhaps MGLMapView should have private methods for each of the things you’d want to do with that object.